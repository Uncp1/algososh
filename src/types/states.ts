export type ElementStatesType =
  | ElementStates.Default
  | ElementStates.Changing
  | ElementStates.Modified;

export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export type LinkedListStateType =
  | LinkedListStates.AddToHead
  | LinkedListStates.AddToTail
  | LinkedListStates.AddByIndex
  | LinkedListStates.DeleteFromHead
  | LinkedListStates.DeleteFromTail
  | LinkedListStates.DeleteByIndex
  | LinkedListStates.Empty
  | LinkedListStates.Success;

export enum LinkedListStates {
  AddToHead = "addToHead",
  AddToTail = "addToTail",
  AddByIndex = "addByIndex",
  DeleteFromHead = "deleteFromHead",
  DeleteFromTail = "deleteFromTail",
  DeleteByIndex = "deleteFromIndex",
  Empty = "null",
  Success = "success",
}
