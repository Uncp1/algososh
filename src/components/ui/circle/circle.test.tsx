import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/states";
import { Circle } from "./circle";

describe("<Circle /> is rendered", () => {
  it("with letters", () => {
    const button = renderer.create(<Circle letter="ABC" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("without letters", () => {
    const button = renderer.create(<Circle />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("with head prop", () => {
    const button = renderer.create(<Circle head={"head"} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("with react element head prop", () => {
    const button = renderer
      .create(<Circle head={<Circle isSmall={true} />} />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });

  it("with tail prop", () => {
    const button = renderer.create(<Circle tail={"tail"} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("with react element tail prop", () => {
    const button = renderer
      .create(<Circle tail={<Circle isSmall={true} />} />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });

  it("with index", () => {
    const button = renderer.create(<Circle index={1} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("with props isSmall ===  true", () => {
    const button = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it(`with state ${ElementStates.Default}`, () => {
    const button = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it(`with state ${ElementStates.Changing}`, () => {
    const button = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it(`with state ${ElementStates.Modified}`, () => {
    const button = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
});
