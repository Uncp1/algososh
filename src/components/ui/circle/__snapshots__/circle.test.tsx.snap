// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Circle /> is rendered with head prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> is rendered with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`<Circle /> is rendered with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> is rendered with props isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> is rendered with react element head prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> is rendered with react element tail prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`<Circle /> is rendered with state changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> is rendered with state default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> is rendered with state modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> is rendered with tail prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`<Circle /> is rendered without letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
